reconcile/utils/terrascript_aws_client.py

```diff
@@ -231,13 +233,37 @@ def __init__(self, integration: str, integration_prefix: str,
             ts += provider.template(
                 version="2.2.0"
             )
-
-            b = Backend("s3",
-                        access_key=config['aws_access_key_id'],
-                        secret_key=config['aws_secret_access_key'],
-                        bucket=config['bucket'],
-                        key=config['{}_key'.format(integration)],
-                        region=config['region'])
+            access_key_value = config['aws_access_key_id']
+            secret_key_value = config['aws_secret_access_key']
+            accounts = queries.get_aws_accounts(terraform_state=True)
+            for filtered_account in filtered_accounts:
+                for account in accounts:
```

Comment:
remember that you are in a loop over `configs` here. everything you do here, you do for each account, e.g. you create the `Backend` filtered_accounts*filtered_accounts times.

have a look at the context here. the variable `name` is bound to the AWS account name your are currently iterating over. so maybe something like the following could work. please verify with a dry run though

```
for acc in filtered_account:
    if acc.get("name") == name:
        # check on terraformState variable in acc (assuming filtered_accounts was created with  terraform_state=True 
        # and create Backend accordingly
```